src/components/CalendarMonthGrid.jsx

```diff
@@ -73,6 +76,23 @@ export default class CalendarMonthGrid extends React.Component {
     this.container.addEventListener('transitionend', this.onTransitionEnd);
   }
 
+  componentWillReceiveProps(nextProps) {
+    const { initialMonth, numberOfMonths } = nextProps;
+    const { months } = this.state;
+
+    const hasMonthChanged = !this.props.initialMonth.isSame(initialMonth, 'month');
+    const hasNumberOfMonthsChanged = this.props.numberOfMonths !== numberOfMonths;
+    if (hasMonthChanged || hasNumberOfMonthsChanged) {
+      if (initialMonth.isAfter(this.props.initialMonth)) {
+        months.shift();
```

Comment:
This is in `componentWillReceiveProps` and is directly mutating the current state which might be a problem, but is doing so based on the change in props not on a change in state.

I don't think we need the callback, but I can figure out a way to do this with `setState` instead of mutating the state under the hood.